### PR TITLE
Add GLM-5.3 to the forecasting recipe and tune defaults / align with epochs

### PR DESCRIPTION
**Pull request by nealwu**
Retune the shared defaults from a 2x3 sweep over both models and 8/16/32 forecasts per question, 128 steps at a 1e-4 learning rate:

* group_size 32. Averaged over steps 64-128, 32 forecasts per question was best for both models (Qwen 0.8223 vs 0.8165/0.8148, GLM 0.8371 vs 0.8359/0.8354) and never worse. It costs little extra wall clock because the forecasts for one question are sampled concurrently: median sampling time was 62.2s at 32 versus 61.4s at 8.
* learning_rate 1e-4 for both models. Four 8e-5 runs spanned 0.8216-0.8261 and the 1e-4 run landed at 0.8223, so the two rates are indistinguishable here; 1e-4 is chosen for consistency across models, not for measured gain.
* max_steps 128 and eval_every 16, matching the runs the results tables come from. One epoch is 64 steps and captures most of the improvement.

GLM-5.3 reaches 0.8475 validation Brier and 78.30% accuracy, vs. 0.8294 and 75.34% for Qwen3.8-27B on the same split.

### CODE REVIEW

File: tinker_cookbook/recipes/forecasting/README.md [hunk 1]
```diff
@@ -102,37 +107,44 @@ and 1,180 questions, which the default caps trim to 1,024 and 256.
...
 
-
 ### Temporal split methodology
+
 This cookbook specifically uses a cutoff of October 20, 2025 UTC, but feel free to adjust based on the model you are using.
 
-* Knowledge cutoff: Try to find the knowledge cutoff for the model you are post-training. Ideally its pretraining cutoff predates the forecasting questions. Qwen3.8 has been reported to self-identify an [early-2025 cutoff](https://huggingface.co/Qwen/Qwen3.8-27B-FP8/discussions/8), although this is not publisher-verified. Adjust the model or split date to reduce outcome contamination.
-* Training split: Events whose markets all closed before the cutoff.
-* Validation split: Events first observed on or after the cutoff.
+- Knowledge cutoff: Try to find the knowledge cutoff for the model you are post-training. Ideally its pretraining cutoff predates the forecasting questions. Qwen3.8 has been reported to self-identify an [early-2025 cutoff](https://huggingface.co/Qwen/Qwen3.8-27B-FP8/discussions/8), although this is not publisher-verified. Adjust the model or split date to reduce outcome contamination.
```
Comment (simonguozirui) [inline tinker_cookbook/recipes/forecasting/README.md:114]: can we mark the same for GLM 5.3 or find sources

Comment (simonguozirui) [reply tinker_cookbook/recipes/forecasting/README.md:114]: actually Z.ai has not publicly disclosed a specific knowledge-cutoff date

File: tinker_cookbook/recipes/forecasting/README.md [hunk 2]
```diff
@@ -4,24 +4,29 @@ Fine-tune a model (Qwen3.8-27B in this example) with Tinker on binary markets fr
...
 python -m tinker_cookbook.recipes.forecasting.data
 
-# train forecasting model
+# Train the forecasting model
 python -m tinker_cookbook.recipes.forecasting.train
+
+# Or train with GLM-5.3 instead
+python -m tinker_cookbook.recipes.forecasting.train \
+    model_name=zai-org/GLM-5.3:peft:262144 \
+    renderer_name=glm5_3_high_reasoning
```
Comment (simonguozirui) [inline tinker_cookbook/recipes/forecasting/README.md:21]: did we do a sweep on various reasoning efforts here? for qwen we find low reasoning to be more efficient and more performant etc

File: tinker_cookbook/recipes/forecasting/README.md [hunk 3]
```diff
@@ -102,37 +107,44 @@ and 1,180 questions, which the default caps trim to 1,024 and 256.
...
 
 **Training loop**
 
-* 16 questions per step
-* 32 forecasts per question
-* 100 training steps
+- 16 questions per step
+- 128 training steps (two passes over the 1,024 training questions)
+
+One epoch is 64 steps. Most of the improvement comes with the first epoch, so
```
Comment (simonguozirui) [inline tinker_cookbook/recipes/forecasting/README.md:146]: awesome this is more clear

**Review by simonguozirui — APPROVED** [review]

**Comment by simonguozirui** [comment]
looks good! small nit on explaining reasoning effort differnece for GLM vs Qwen etc, would be good to give a rationale

File: tinker_cookbook/recipes/forecasting/README.md (hunk 2 above)
Comment (nealwu) [reply tinker_cookbook/recipes/forecasting/README.md:21]: added a few notes about this to the README, further down
